src/classes/User/User.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { User } from './User';
+import { UsersApi, UserInfo } from '../../generated';
+import { apiConfiguration } from '../../utils/apiConfiguration';
+
+// Mock the generated APIs
+jest.mock('../../generated');
+
+describe('User', () => {
+    let user: User;
+    const mockConfig = {
+        appId: 'test-app-id',
+        apiConfiguration: apiConfiguration(),
+    };
+
+    const mockUserInfo: Partial<UserInfo> = {
+        id: 'user-123',
+        createdAt: new Date('2024-01-01'),
+        lastLoginAt: new Date('2024-01-02'),
+        loginCount: 5,
+        status: 'active',
+        updatedAt: new Date('2024-01-02'),
+        userMetadata: {},
+        webauthn: true,
+        webauthnDevices: [],
+        webauthnTypes: [],
+    };
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        user = new User(mockConfig);
+    });
+
+    describe('get', () => {
+        it('should successfully map user info', async () => {
+            // Mock the API responses
+            (UsersApi.prototype.listPaginatedUsers as jest.Mock).mockResolvedValue({
+                users: [{ id: 'user-123' }],
+            });
+            (UsersApi.prototype.getUser as jest.Mock).mockResolvedValue({
+                user: mockUserInfo,
+            });
+
+            const result = await user.get('external-123');
+
+            // Verify result
+            expect(result).toEqual({
+                id: 'user-123',
+                createdAt: new Date('2024-01-01'),
+                lastLoginAt: new Date('2024-01-02'),
+                loginCount: 5,
+                status: 'active',
+                updatedAt: new Date('2024-01-02'),
+                userMetadata: {},
+                webauthn: true,
+                webauthnDevices: [],
+                webauthnTypes: [],
+            });
+        });
+
+        it('should throw error when user is not found', async () => {
+            // Mock empty user list response
+            (UsersApi.prototype.listPaginatedUsers as jest.Mock).mockResolvedValue({
+                users: [],
+            });
+
+            await expect(user.get('non-existent-id')).rejects.toThrow('Could not find user with that external ID');
```

Comment:
i was going to suggest we change this to include the external id, but since it could be PII, we probably shouldn't. is that your understanding as well?

Reply:
I didn't really think too much about this error case, this was the error we're already throwing. You'll only get this error if you tried querying an external ID so you'd have access to that information at this point in your server code anyways.

Reply:
I was thinking about how someone would likely log this out in the handling of the error, where it could be less likely they log out their external identifier before this.

either way, all good without any changes; just wanted to get your opinion on it. feel free to merge!